api/pkg/instrumentation/metrics/nop.go

```diff
@@ -0,0 +1,26 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"time"
+)
+
+// NopMetricsCollector implements the Collector interface with a set of
+// Nop methods
+type NopMetricsCollector struct {
+}
+
+func newNopMetricsCollector() Collector {
+	return &NopMetricsCollector{}
+}
+
+// InitMetrics satisfies the Collector interface
+func (NopMetricsCollector) InitMetrics(map[MetricName]*prometheus.HistogramVec) {}
```

Comment:
Referring to the Prometheus library in this file doesn't quite make sense, right? In the future, for esample, there can be a `Statsd` Metrics collector. `metrics.go` and `nop.go` should not be concerned with individual collectors related functionality.